app/src/components/common/TmSessionImportName.vue

```diff
@@ -99,29 +113,82 @@ export default {
         this.$store.commit(`updateField`, { field: `name`, value })
       },
     },
+    networkAccountTypes() {
+      return JSON.parse(this.currentNetwork.accountTypes)
+    },
+    currentAlgo() {
+      return this.networkAccountTypes[this.attempt]
+    },
   },
-  async created() {
-    this.importedAddress = await this.$store.dispatch(`getAddressFromSeed`, {
-      seedPhrase: this.$store.state.recover.seed,
-      network: this.network,
-    })
+  mounted() {
+    this.created()
   },
   methods: {
     onSubmit() {
       this.$v.$touch()
       if (this.$v.name.$invalid) return
       this.$router.push("/recover/password")
     },
+    async created(retry = false) {
+      if (retry) this.attempt++
+      this.attempt = this.numberAttemptsController(
+        JSON.parse(this.currentNetwork.accountTypes),
+        this.attempt
+      )
+      const wallet = await this.$store.dispatch(`getAddressFromSeed`, {
+        seedPhrase: this.$store.state.recover.seed,
+        network: this.network,
+        attempt: this.attempt,
+      })
+      this.importedAddress = wallet.cosmosAddress
+      this.$store.commit(`updateField`, {
+        field: `accountType`,
+        value: this.currentAlgo,
+      })
+    },
+    numberAttemptsController(accountTypes, attempt) {
+      if (attempt >= accountTypes.length) {
+        return attempt - accountTypes.length
+      } else {
+        return attempt
+      }
+    },
   },
   validations: () => ({
     name: { required, minLength: minLength(3), nameExists },
   }),
 }
 </script>
-<style scoped>
+<style>
```

Comment:
why did you remove the scope?

Reply:
I'm still struggling with the class for the algo span... the `TmFormGroup` component is enforcing its own classes. I thought with this I will fix it but yesterday evening it wasn't working yet

Reply:
Ah no, it seems it is working now

![image](https://user-images.githubusercontent.com/40721795/87316531-31040e80-c526-11ea-961a-75467e188198.png)

Reply:
@jbibla can you help?

Reply:
It is OK, I think this is the way to fix it. Right @jbibla ?

Reply:
please add scoped back